sql-plugin/src/main/scala/com/nvidia/spark/rapids/AstUtil.scala

```diff
@@ -119,4 +124,169 @@ object AstUtil {
+   * @param left      original expressions from join's left child. It's left project input
+   *                  attribute.
+   * @param leftProj  extra expressions extracted from original join condition which is not
+   *                  supported by ast. It will be evaluated as a project on left side batch.
+   * @param right     original expressions from join's right child. It's left project input
+   *                  attribute.
+   * @param rightProj extra expressions extracted from original join condition which is not
+   *                  supported by ast. It will be evaluated as a project on right side batch.
+   * @param post      eliminate the extra columns introduced by join condition split
+   * @param buildSide indicates which side is build
+   */
+  case class JoinCondSplitAsProject(
+      astCond: Option[Expression],
+      left: Seq[NamedExpression], leftProj: Seq[NamedExpression],
+      right: Seq[NamedExpression], rightProj: Seq[NamedExpression],
+      post: Seq[NamedExpression], buildSide: GpuBuildSide
+  ) extends JoinCondSplitStrategy(left, right, buildSide) {
+    private[this] val leftInput = left.toSeq.map(_.toAttribute)
+    private[this] val rightInput = right.toSeq.map(_.toAttribute)
+
+    // Used to build build/stream side project
+    private[this] val (buildOutput, streamOutput, buildInput, streamInput) = buildSide match {
+      case GpuBuildLeft =>
+        (leftOutput, rightOutput, leftInput, rightInput)
+      case GpuBuildRight =>
+        (rightOutput, leftOutput, rightInput, leftInput)
+    }
+
+    private[this] val buildProj = if (!buildOutput.isEmpty) {
+      Some(GpuBindReferences.bindGpuReferencesTiered(buildOutput.toSeq, buildInput.toSeq, false))
+    } else None
+
+    private[this] val streamProj = if (!streamOutput.isEmpty) {
+      Some(GpuBindReferences.bindGpuReferencesTiered(streamOutput.toSeq, streamInput.toSeq, false))
+    } else None
+
+    // Remove the intermediate attributes from left and right side project nodes. Output attributes
+    // need to be updated based on join type. And its attributes covers both original plan and
+    // extra project node.
+    private[this] val postProj = if (!post.isEmpty) {
+      Some(
+        GpuBindReferences.bindGpuReferencesTiered(
+          post.toSeq, (leftOutput ++ rightOutput).map(_.toAttribute).toSeq, false))
+    } else None
+
+    override def astCondition(): Option[Expression] = astCond
+
+    // This is the left side child of join. In `split as project` strategy, it may be different
+    // from original left child with extracted join condition attribute.
+    override def leftOutput(): Seq[NamedExpression] = left ++ leftProj
+
+    // This is the right side child of join. In `split as project` strategy, it may be different
+    // from original right child with extracted join condition attribute.
+    override def rightOutput(): Seq[NamedExpression] = right ++ rightProj
```

Comment:
Good suggestion. Updated.

Reply:
The latter part may be a little bit risky in current case around the class value member initialization order. Since there exists [other variables](https://github.com/NVIDIA/spark-rapids/pull/9760/files#diff-6d842883ea07751fad4e05bd48a8aa750f25319b74a65b1a200938b4c2866addR141-R144 ) needs ```leftOutput``` and ```rightOutput```, order matters here. Just made it a method as passing as ```JoinCondSplitStrategy``` parameter already simplify here.